crates/common/account_manager/src/lib.rs

```diff
@@ -1,27 +1,51 @@
+use std::str::FromStr;
+
+use bip39::Mnemonic;
 use rand::SeedableRng;
 use rand_chacha::ChaCha20Rng;
 use ream_post_quantum_crypto::hashsig::{private_key::PrivateKey, public_key::PublicKey};
 use sha2::{Digest, Sha256};
 use tracing::info;
 
+pub mod keystore;
+pub mod message_types;
+pub mod utils;
+
 pub fn generate_keys(
     seed_phrase: &str,
-    activation_epoch: usize,
-    num_active_epochs: usize,
+    wallet_index: u32,
+    activation_epoch: u32,
+    num_active_epochs: u32,
+    passphrase: &str,
 ) -> (PublicKey, PrivateKey) {
     info!(
-        "Generating lean consensus validator keys  with activation_epoch={activation_epoch}, num_active_epochs={num_active_epochs}....."
+        "Generating lean consensus validator keys for index {wallet_index} with activation_epoch={activation_epoch}, num_active_epochs={num_active_epochs}....."
     );
 
-    // Hash the seed phrase to get a 32-byte seed
+    // Parse the mnemonic phrase
+    let mnemonic = Mnemonic::from_str(seed_phrase).expect("Invalid mnemonic phrase");
+
+    // Generate seed from mnemonic using provided passphrase
+    let seed = mnemonic.to_seed(passphrase);
+
+    // Create a deterministic seed based on the original seed and wallet index
     let mut hasher = Sha256::new();
-    hasher.update(seed_phrase.as_bytes());
-    let seed = hasher.finalize().into();
-    info!("Seed: {seed:?}");
-
-    PrivateKey::generate(
-        &mut <ChaCha20Rng as SeedableRng>::from_seed(seed),
-        activation_epoch,
-        num_active_epochs,
-    )
+    hasher.update(seed);
+    hasher.update(wallet_index.to_be_bytes());
+    let derived_seed: [u8; 32] = hasher.finalize().into();
+
+    // Use the derived seed directly for hashsig key generation
+    let (public_key, private_key) = PrivateKey::generate(
+        &mut <ChaCha20Rng as SeedableRng>::from_seed(derived_seed),
+        activation_epoch as usize,
+        num_active_epochs as usize,
+    );
+
+    // Display public key contents
+    match serde_json::to_string_pretty(&public_key.inner) {
+        Ok(json) => info!("Public key contents: {json}"),
+        Err(e) => info!("Public key generated successfully (could not serialize) due to {e}"),
```

Comment:
```suggestion
        Err(err) => info!("Public key generated successfully (could not serialize) due to {err}"),
```

for consistency

Reply:
Another standard that should added/recommend to be followed within the codebase. `No single letter variables`. @KolbyML